mono/mini/tramp-amd64.c

```diff
@@ -363,7 +363,12 @@ mono_arch_create_generic_trampoline (MonoTrampolineType tramp_type, MonoTrampInf
 	}
 	for (i = 0; i < AMD64_XMM_NREG; ++i)
 		if (AMD64_IS_ARGUMENT_XREG (i))
+#if defined(__APPLE__) || (defined(__linux__) && defined(__GLIBC__)) || defined(HOST_WIN32)
```

Comment:
Could we move this to mono-context as a macro that is defined when MonoContextSimdReg is present?
IE, have it define `MONO_HAVE_SIMD_REG` and this code consume it.

Reply:
It's doable, but I'm uncertain to how far the checks go. I'll work on this and push another commit.